File: Models/LanguageOption.vb
Namespace Models
    Public Class LanguageOption
        Private ReadOnly _cultureName As String
        Private ReadOnly _nativeName As String
        Private ReadOnly _englishName As String

        Public Sub New(cultureName As String, nativeName As String, englishName As String)
            _cultureName = If(cultureName, String.Empty)
            _nativeName = If(nativeName, String.Empty)
            _englishName = If(englishName, String.Empty)
        End Sub

        Public ReadOnly Property CultureName As String
            Get
                Return _cultureName
            End Get
        End Property

        Public ReadOnly Property NativeName As String
            Get
                Return _nativeName
            End Get
        End Property

        Public ReadOnly Property EnglishName As String
            Get
                Return _englishName
            End Get
        End Property

        Public ReadOnly Property DisplayName As String
            Get
                If String.IsNullOrWhiteSpace(NativeName) Then
                    Return EnglishName
                End If

                If String.IsNullOrWhiteSpace(EnglishName) OrElse String.Equals(NativeName, EnglishName, StringComparison.OrdinalIgnoreCase) Then
                    Return NativeName
                End If

                Return NativeName & " / " & EnglishName
            End Get
        End Property

        Public Overrides Function ToString() As String
            Return DisplayName
        End Function
    End Class
End Namespace

File: src/ClickSyncMouseTester/Models/LanguageOption.cs
using System;

namespace ClickSyncMouseTester.Models;

public class LanguageOption
{
    private readonly string _cultureName;

    private readonly string _nativeName;

    private readonly string _englishName;

    public string CultureName => _cultureName;

    public string NativeName => _nativeName;

    public string EnglishName => _englishName;

    public string DisplayName
    {
        get
        {
            if (string.IsNullOrWhiteSpace(NativeName))
            {
                return EnglishName;
            }
            if (string.IsNullOrWhiteSpace(EnglishName) || string.Equals(NativeName, EnglishName, StringComparison.OrdinalIgnoreCase))
            {
                return NativeName;
            }
            return NativeName + " / " + EnglishName;
        }
    }

    public LanguageOption(string cultureName, string nativeName, string englishName)
    {
        _cultureName = cultureName ?? string.Empty;
        _nativeName = nativeName ?? string.Empty;
        _englishName = englishName ?? string.Empty;
    }

    public override string ToString()
    {
        return DisplayName;
    }
}





